### PR TITLE
Apply Dart 3 patterns on Flame codelab

### PR DESCRIPTION
**Pull request by miquelbeltran**
Changes on the codelab document will be required

## Pre-launch Checklist

- [ ] I read the [Effective Dart: Style] _recently_, and have followed its advice.
- [ ] I signed the [CLA].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-devrel channel on [Discord].

[Effective Dart: Style]: https://dart.dev/guides/language/effective-dart/style
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by miquelbeltran** [comment]
@domesticmouse I'll need access to the codelab document to make the code changes there

**Comment by domesticmouse** [comment]
> @domesticmouse I'll need access to the codelab document to make the code changes there

shared the doc with you

File: flame-building-doodle-dash/finished_game/pubspec.yaml [hunk 1]
```diff
@@ -19,7 +19,7 @@ publish_to: 'none' # Remove this line if you wish to publish to pub.dev
 version: 1.0.0+1
 
 environment:
-  sdk: '>=3.0.0 <4.0.0'
+  sdk: '>=3.0.2 <4.0.0'
```
Comment (kevmoo) [inline flame-building-doodle-dash/finished_game/pubspec.yaml:22]: Could be `^3.0.2` – although constraining to patch releases seems a bit too much...

Comment (domesticmouse) [reply flame-building-doodle-dash/finished_game/pubspec.yaml:22]: This is what ever `flutter create` writes out.

**Review by domesticmouse — APPROVED** [review]

**Comment by miquelbeltran** [comment]
I've merged in your changes from https://github.com/flutter/codelabs/pull/1635 to avoid merge issues later on.

The codelab document is updated now, thanks for sharing it!

As noted on https://github.com/flutter/flutter/issues/127779 , upgrading the codelab to Flame 1.8.0 is more complex as I thought, since the deprecated API and the new one don't really match 1:1.

**Comment by domesticmouse** [comment]
> I've merged in your changes from https://github.com/flutter/codelabs/pull/1635 to avoid merge issues later on.
> 
> 
> 
> The codelab document is updated now, thanks for sharing it!
> 
> 
> 
> As noted on https://github.com/flutter/flutter/issues/127779 , upgrading the codelab to Flame 1.8.0 is more complex as I thought, since the deprecated API and the new one don't really match 1:1.

I think I'll add a note to the codelab about using Flame 1.7.3. That way we aren't time locked to updating to 1.8.0

**Comment by domesticmouse** [comment]
> I think I'll add a note to the codelab about using Flame 1.7.3. That way we aren't time locked to updating to 1.8.0

This update is held up by internal tooling issues.

Googlers: b/284553092

**Comment by kevmoo** [comment]
still blocked?

**Comment by domesticmouse** [comment]
Yup. And it's all gDoc API's fault.

**Pull request merged**